xarray/core/combine.py

```diff
@@ -499,10 +499,12 @@ def _auto_combine_1d(datasets, concat_dim=_CONCAT_DIM_DEFAULT,
     # This is just the old auto_combine function (which only worked along 1D)
     if concat_dim is not None:
         dim = None if concat_dim is _CONCAT_DIM_DEFAULT else concat_dim
-        grouped = itertools.groupby(datasets, key=lambda ds: tuple(sorted(ds)))
+        sorted_datasets = sorted(datasets, key=lambda ds: tuple(sorted(ds)))
```

Comment:
nit: define a single key function and reuse it here and below, e.g., `key_func = lambda ds: tuple(sorted(ds))`